templates/openstack-userdata.sh.j2

```diff
@@ -97,7 +97,6 @@ write_post_metrics(){
 }
 
 # Run runner
-su - ubuntu -c "cd ~/actions-runner && /home/ubuntu/actions-runner/run.sh"
-write_post_metrics $?
+(set +e; su - ubuntu -c "cd ~/actions-runner && /home/ubuntu/actions-runner/run.sh"; write_post_metrics $?)
```

Comment:
Is this to ensure the `write_post_metrics` only execute on the `run.sh` completed successfully?
Would `.../run.sh && write_post_metrics $?` be more readable?
Would it be good to add a comment to explain it?

Reply:
No, this is to ensure that `write_post_metrics` runs even if run.sh fails. I could simplify this by using the `||` operator.

Reply:
Oh it is `set +e` and not `set -e`. Still would it is better as `.../run.sh || write_post_metrics`, or adding a comment?

Reply:
Using `||` would prevent write_post_metrics from running in a happy case. I'll add a comment.